### PR TITLE
163148392 detect multiple change times per route v2

### PR DESCRIPTION
**Pull request by erno**
Supersedes #889

### CODE REVIEW

File: ote/src/cljc/ote/time.cljc [hunk 1]
```diff
@@ -153,9 +153,13 @@
 (defn date-fields->date-time [{::keys [year date month hours minutes seconds]}]
   (t/date-time year month date hours minutes seconds))
 
+(defn date-fields->date-midnight [{::keys [year date month]}]
+  #_(t/date-midnight year month date)
+  (t/date-time year month date))
+
 (defn date-fields->date [{::keys [year date month]}]
   #?(:clj (java.time.LocalDate/of year month date)
-     :cljs (goog.date.Date. year (dec month) date)))
```
Comment (erno) [inline ote/src/cljc/ote/time.cljc]: Is this an intentional change?

File: ote/test/clj/ote/transit_changes/detection_test.clj [hunk 2]
```diff
@@ -169,10 +173,302 @@
...
+                               :different-week
+                               :beginning-of-week))))
+
+    (testing "second change is detected"
+      (is (= (d 2019 3 4) (-> diff-pairs
+                              second
+                              :different-week
+                              :beginning-of-week))))))
+
+
```
Comment (erno) [inline ote/test/clj/ote/transit_changes/detection_test.clj]: there are now several defined test datas that are not unused, do we need to add more tests to use them or should unused data be removed?

File: ote/test/clj/ote/transit_changes/detection_test.clj [hunk 3]
```diff
@@ -169,10 +173,302 @@
...
+  ;; first test that the test data and old change detection code agree
+  (testing "single-change detection code agrees with test data"
+    (is (= (d 2019 2 18) (-> test-more-than-one-change
+                             detection/route-weeks-with-first-difference
+                             (get route-name)
+                             :different-week
+                             :beginning-of-week))))
+  
+  (let [diff-pairs (-> test-more-than-one-change
```
Comment (erno) [inline ote/test/clj/ote/transit_changes/detection_test.clj]: general "diff-pairs" naming comment, the data format isn't pairs anymore I think, so should globally rename these

File: ote/src/cljc/ote/time.cljc [hunk 4]
```diff
@@ -411,6 +415,12 @@
   [native-date]
   (date-fields->date-time (date-fields native-date)))
 
+(defn native->date
```
Comment (philosotractor) [inline ote/src/cljc/ote/time.cljc]: This is not used ATM

**Pull request merged**